packages/examples/kura/integration_with_kura.py

```diff
@@ -0,0 +1,256 @@
+# ----------------------------------------------------------------------------#
+
+import logging
+import json
+import time
+import threading
+import random
+import copy
+from liota.entities.devices.simulated_device import SimulatedDevice
+from liota.entities.metrics.metric import Metric
+from liota.device_comms.mqtt_device_comms import MqttDeviceComms
+from liota.core.package_manager import LiotaPackage
+from liota.lib.utilities.utility import get_default_network_interface, get_disk_name
+
+
+log = logging.getLogger(__name__)
+dependencies = ["iotcc_mqtt"]
+network_interface = get_default_network_interface()
+disk_name = get_disk_name()
+no_of_edge_system_in_thousands = 1
+# Number of Retries for Connection and Registrations
+no_of_retries_for_connection = 5
+# Retry delay Min Value in seconds
+delay_retries_min = 600
+# Retry delay Max Value in seconds
+delay_retries_max = 1800
+
+# Lambda Function Multiplier uses the above settings for calculating retry and delay logic
+lfm = lambda x: x * no_of_edge_system_in_thousands
+retry_attempts = lfm(no_of_retries_for_connection)
+delay_retries = random.randint(lfm(delay_retries_min), lfm(delay_retries_max))
+
+Msg_dic = {}                #{"demo": {"flag": 0, "msg": 0, "metric": metric}}
+Gatewaydict = {}            #{"assetName": reg_device}
+
+callbackFunc_str = '''def metric_callback_%s():    return Msg_dic['%s']['msg']''' # update the data by use Msg_dic disc
+
+"""Subscibe callback function"""
+def sub_callback(self, client, userdata, msg):
+
+    if (msg.topic.split("/").__len__() == 4) and (msg.topic.split("/")[3] == "pulse-topic"): # msg.topic = "pulse-account-name/pulseid/pulse/pulse-topic"
+
+        Jsonpayload = json.loads(msg.payload)['metrics']
+        if Jsonpayload: # msg.payload = "{"sentOn":1533007387525,"metrics":{"assetName":"Store-GW-03","sensor_timestamp":1533007387525,"sensor":0}}"
+
+            #is new gateway?
+            if Jsonpayload['assetName'] in Gatewaydict.keys():
+
+                log.info("Device already registered.")
+
+                for key_metric in Jsonpayload.keys():
+
+                    # filter the unuseful sensor
+                    if key_metric == "assetName":
+                        continue
+                    if key_metric.find('stamp') >= 0:
+                        continue
+
+                    if key_metric in Msg_dic.keys(): #exist update msg
+
+                        Msg_dic[key_metric]['msg'] = Jsonpayload[key_metric]
+                        Msg_dic[key_metric]['flag'] = 0
+
+                    else:
+
+                        Msg_dic[key_metric] = {}
+                        Msg_dic[key_metric]['msg'] = Jsonpayload[key_metric]
+                        Msg_dic[key_metric]['flag'] = 1
+
+                        exec(callbackFunc_str % (key_metric, key_metric)) # Dynamically generating callback function
+
+                        try:
+                            metric_ins = Metric(name=key_metric,
```

Comment:
We need to discuss this approach for every metric are you creating the collecting cycle or will it be done only once?